src/main/java/org/ibp/api/java/impl/middleware/manager/UserValidator.java

```diff
@@ -231,22 +219,157 @@ protected void validatePersonEmailIfExists(final Errors errors, final String eMa
 			if (null == errors.getFieldError(EMAIL) && this.userService.isPersonWithEmailExists(eMail)) {
 				errors.rejectValue(EMAIL, SIGNUP_FIELD_EMAIL_EXISTS);
 			}
-		} catch (MiddlewareQueryException e) {
+		} catch (final MiddlewareQueryException e) {
 			errors.rejectValue(EMAIL, DATABASE_ERROR);
 			LOG.error(e.getMessage(), e);
 		}
 	}
 
-    protected void validateUserRole(final Errors errors, final Role role) {
-		if (null == role) {
-            errors.rejectValue(ROLE, SIGNUP_FIELD_INVALID_ROLE);
-		} else if (this.isSuperAdminRole(role)) {
-			errors.reject(CANNOT_ASSIGN_SUPERADMIN_ROLE);
-		} else {
-	        this.validateFieldLength(errors, role.getDescription(), ROLE, ROLE_STR, 30);
-		} 
+	protected void validateUserRoles(final Errors errors, final UserDetailDto user) {
+
+		final List<UserRoleDto> userRoles = user.getUserRoles();
+
+		if (!userRoles.isEmpty()) {
+
```

Comment:
This is one very long method. We can extract related validations to separate methods, say validation related to roles, crop, role types, etc